### ISSUE TITLE
HandleCursorShape and GotCursorShapeProc not adequately documented

### ISSUE THREAD

**Issue by willbprog127** [LibVNC/libvncserver#93]
First off, thank you so much for libvncserver/client...it's been very helpful!

`HandleCursorShape` and `GotCursorShapeProc` are not documented adequately.  They are listed in the documentation, but there are no helpful remarks accompanying them.  I have no clue how to use them, nor can I find any real usage examples, even when searching the internet.

```
rfbBool HandleCursorShape 	( 	rfbClient *  	client,
		int  	xhot,
		int  	yhot,
		int  	width,
		int  	height,
		uint32_t  	enc 
	) 		

Definition at line 46 of file cursor.c.
```
```
typedef void(* GotCursorShapeProc)(struct _rfbClient *client, int xhot, int yhot, int width, int height, int bytesPerPixel)

Definition at line 173 of file rfbclient.h.
```

Can you please add useful information to the documentation for these functions so people can actually know how to implement them?  Without using them, I see no cursor changes, even though the VNC server I am contacting is sending them (as seen in other VNC viewers light TightVNC, TigerVNC, etc.)

Thank you! :smile:

**Comment by bk138** [comment]
See 4665af4950023c194453fe7517dbe06ee811ea15. Note that you can also set your client to not handle cursor shape updates, thus the server will draw the cursor into the framebuffer.

**Comment by willbprog127** [comment]
> See 4665af4. Note that you can also set your client to not handle cursor shape updates, thus the server will draw the cursor into the framebuffer.

Thanks for that, and for changing the documentation.  When you say *"you can also set your client to not handle cursor shape updates"*, do you do it this way?...
```
client->GotCursorShape = 0;
```
I have had no success in the cursor changing, even though the server is sending the changes.  For testing, I set up the callback to handle the cursor shape change and it *is* getting called when I do...
```
client->appData.useRemoteCursor = true;
```
Thanks :-)

**Comment by bk138** [comment]
I hope 9d4cb568b70443f9137108dd5eaad34523c3664d explains this.

**Comment by willbprog127** [comment]
@bk138 Thanks for that.  I will check it out, and thank you again for clarifying in the documentation. :+1:

**Comment by willbprog127** [comment]
I looked at how `SetFormatAndEncodings` works, and it forces `rfbEncodingXCursor` and so-on to be sent when I set `useRemoteCursor` to true.  

I've tried every different mix of `useRemotecursor` and `GotCursorShape` that I can think of.  I'm not seeing any cursor changes on the framebuffer and when I try to handle it with the `GotCursorShapeProc` callback, `client->rcSource` and `client->rcMask` are invalid.

Anything else I should be doing to see cursor changes?

Thanks, and sorry if I'm not 'getting it'.  RFB isn't my forte'.
